src/util/aux_util.h

```diff
@@ -28,7 +28,7 @@ extern "C" {
 #endif
 
 #define UNUSED(x) (void)(x)
-#if (MAXLOGLEVEL == LOGL_NONE) || defined(NDEBUG)
+#if (MAXLOGLEVEL == LOGL_NONE) || !defined(NDEBUG)
```

Comment:
That is wrong. If NDEBUG is defined then we do want to have have MAYBE_UNUSED defined also.

Reply:
> That is wrong. If NDEBUG is defined then we do want to have have MAYBE_UNUSED defined also.

Because debug builds reference things for debug output that release builds wont, got it. But since I was able to make changes, combined we have done NDEBUG and non NDEBUG builds successfully, which *i think* means NDEBUG case ins't needed. Testing this hypothesis now.